python/plugins/processing/algs/qgis/FieldPyculator.py

```diff
@@ -59,16 +61,31 @@ def __init__(self):
         super().__init__()
 
     def initAlgorithm(self, config=None):
-        self.type_names = [self.tr('Integer'),
-                           self.tr('Float'),
-                           self.tr('String')]
-
         self.addParameter(QgsProcessingParameterFeatureSource(self.INPUT, self.tr('Input layer'),
                                                               types=[QgsProcessing.TypeVector]))
         self.addParameter(QgsProcessingParameterString(self.FIELD_NAME,
                                                        self.tr('Result field name'), defaultValue='NewField'))
-        self.addParameter(QgsProcessingParameterEnum(self.FIELD_TYPE,
-                                                     self.tr('Field type'), options=self.type_names))
+
+        types = [[QVariant.Int, QVariant.Invalid],
+                 [QVariant.Double, QVariant.Invalid],
+                 [QVariant.String, QVariant.Invalid],
+                 [QVariant.Bool, QVariant.Invalid],
+                 [QVariant.Date, QVariant.Invalid],
+                 [QVariant.Time, QVariant.Invalid],
+                 [QVariant.DateTime, QVariant.Invalid],
+                 [QVariant.ByteArray, QVariant.Invalid],
+                 [QVariant.StringList, QVariant.String],
+                 [QVariant.List, QVariant.Int],
+                 [QVariant.List, QVariant.Double]]
+        type_names = []
+        type_icons = []
+        for t in types:
+            type_names.append(QgsVariantUtils.typeToDisplayString(t[0], t[1]))
+            type_icons.append(QgsFields.iconForFieldType(t[0], t[1]))
```

Comment:
```suggestion
        for type_name, subtype_name in types:
            type_names.append(QgsVariantUtils.typeToDisplayString(type_name, subtype_name))
            type_icons.append(QgsFields.iconForFieldType(type_name, subtype_name))
```

Because python ;-P